service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java

```diff
@@ -443,6 +446,77 @@ public void listValidRegions() throws Exception {
     assertTrue(azures.isEmpty());
   }
 
+  @Test
+  @EnabledIf(expression = "${feature.tps-enabled}", loadContext = true)
+  public void updatePolicies_tpsEnabledAndPolicyUpdated() throws Exception {
+    ApiWorkspaceDescription workspaceWithoutPolicy =
+        mockMvcUtils.getWorkspace(userAccessUtils.defaultUserAuthRequest(), workspace.getId());
+    assertTrue(workspaceWithoutPolicy.getPolicies().isEmpty());
+
+    // add attributes
+    String usRegion = "US";
```

Comment:
This looks ok, but I'm not sure why we have capital "US" here. That's not one of the regular regions. And we use lower case "us" on 487. Maybe we should use the "usa" region?

Reply:
it does allow me to update policy attributes to US though. i thought it's tps being smart. is it a bug?

Reply:
We don't currently validate the value of the constraint. So you can set it to US if you want. If you attempt to merge that policy with something later or retrieve valid locations for it, then you won't be able to merge any policies or find any valid locations.
Maybe we should be validating that input.

Reply:
agree with validating the input.